.github/workflows/build_and_push_staging.yml

```diff
@@ -0,0 +1,81 @@
+name: Docker build and push to staging
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  GITHUB_SHA: ${{ github.sha }}
+  AWS_ACCOUNT_STAGING: 843973686572
+  REGISTRY: ${{env.AWS_ACCOUNT_STAGING}}.dkr.ecr.ca-central-1.amazonaws.com/url-shortener/api
```

Comment:
I'm not sure if this will work - I seem to remember that you can't reference `env` values from `env` blocks.  But, I could also be making this up so more than happy to try it as-is to see what happens!   🧪

Reply:
Ok, let's try it! I could not find any definitive information on this, so let's see what happens.